framework/contracts/native/ans-host/src/contract.rs

```diff
@@ -103,13 +103,29 @@ pub fn query(deps: Deps, env: Env, msg: QueryMsg) -> StdResult<Binary> {
 }
 
 #[cfg_attr(feature = "export", cosmwasm_std::entry_point)]
-pub fn migrate(deps: DepsMut, _env: Env, _msg: MigrateMsg) -> AnsHostResult {
-    let version: Version = CONTRACT_VERSION.parse().unwrap();
+pub fn migrate(deps: DepsMut, env: Env, msg: MigrateMsg) -> AnsHostResult {
+    match msg {
+        MigrateMsg::Instantiate(instantiate_msg) => {
+            let contract_info = deps
+                .querier
+                .query_wasm_contract_info(&env.contract.address)?;
+            // Only admin can call migrate on contract
+            let sender = contract_info.admin.unwrap();
```

Comment:
This is ensured by the VM no?

Reply:
Yeah, it's just for construction of `MessageInfo`, in case instantiate will need "sender"

Reply:
Can we pull this logic into a function in abstract-std?

Reply:
Something like `fn migrate_instantiation(init_fn: Fn<(...)> -> ...)`

Reply:
> Something like `fn migrate_instantiation(init_fn: Fn<(...)> -> ...)`

Yep